Refactor the teraranger driver: uniform initialization, format whitespace, deprecate usage of the ringbuffer, etc.

mcsauder opened the pull request:
**Describe problem solved by the proposed pull request**
This PR refactors the teraranger distance sensor driver.  Usage of the ringbuffer is deprecated. Uniform initialization is employed along with formatting/alphabetizing/organizing, and standardizing  the driver against other current distance sensor driver implementations.

**Describe your preferred solution**
Standardizing all of the distance sensor driver variable initialization, method ordering and general style will allow for future inheritance structure work to be performed on the distance sensor driver classes.

**Describe possible alternatives**
All of the distance sensor driver work could be accomplished in one massive PR, but breaking up work in each driver should minimize risk and reduce review effort.

**Additional context**
See #9279, #11853, #11857, #1858, #11859, #11891

Please let me know if you have any questions on this PR. Thanks!

-Mark

mcsauder commented:
@msadowski , would you be willing to review this PR for me?  I am working to align the distance sensor drivers so that an inheritance hierarchy can be created.  Thanks for any feedback you might have!

msadowski commented:
@mcsauder great effort on reducing code duplication! I had a read through and the changes make sense to me. Unfortunately I don't have access to TeraRanger sensors anymore but I'll forward this issue to some people from Terabee so that they can verify the changes work as expected with the sensors.

mcsauder commented:
Thank you @msadowski !

msadowski commented:
@mcsauder I just got my hands on some TeraRanger sensors. Will be testing them with your PR in the next couple of days and will get back to you on how they work.

mcsauder commented:
thanks @msadowski ! I finally got hardware as well!  Bench testing this PR on hardware checks out, I see no differences between this PR and PX4:master.

Current master bench test:
![image](https://user-images.githubusercontent.com/2497951/60987721-6ec0cc80-a2ff-11e9-9f3a-8412a1e2c0bd.png)

This PR:
![image](https://user-images.githubusercontent.com/2497951/60987815-9fa10180-a2ff-11e9-8dd0-7c249a1ce87b.png)

mcsauder commented:
Hi @msadowski , I've added another commit deprecating IOCTL usage and doing a massive re-aligment of this driver to match other driver work I am putting together in #11977.  

Here is a screen capture of the most recent commit, it's performing identically to master:
![image](https://user-images.githubusercontent.com/2497951/60989914-6919b580-a304-11e9-98b5-eba774089874.png)

msadowski commented:
@mcsauder I just tested your changes with Pixhawk 1 and TeraRanger One and so far so good! I'll be testing the following sensors:
* TeraRanger One
* TeraRanger Evo 600Hz
* TeraRanger Evo 60M
* TeraRanger Evo 3M

If I find any issues I'll let you know. I see that the 3M ranges have not been added yet to the driver. Would you mind if I add them and make a PR to your teraranger_driver_work branch?

msadowski commented:
One thing that I noticed happening now is that the DISTANCE_SENSOR fields:
* horizonta_fov
* vertical_fov
* quaternion.0-3
are being filled with what looks like garbage data (will investigate it further soon!):
![image](https://user-images.githubusercontent.com/6543766/61111967-18809480-a48b-11e9-98e2-562c615dcaf8.png)

mcsauder commented:
Hi @msadowski , thanks so much for your testing and feedback.

I've pushed up your commit merged into my branch, and repaired breakage I inadvertently created with the test() method.

Nice catch on the garbage data, I see this in PX4:master branch as well so it is not new to this PR:
![image](https://user-images.githubusercontent.com/2497951/61142689-99d92680-a48d-11e9-834c-9c4fed3e22af.png)

msadowski commented:
@mcsauder Thanks a lot for applying my changes.

I just tested your latest commit with TeraRanger One, Evo 600Hz, Evo 60M and Evo 3M with the following boards: Pixhawk 1, Pixhawk Cube and Pixhawk 4.

Everything seems to be working correctly from my side!

mcsauder commented:
Thanks @msadowski for all of your testing!

@dagar , rebased against current master and good to go.

mcsauder commented:
Hi @dagar , I've deprecated the ringbuffer usage as well as the `test()` and `read()` methods, (because `read()` relied on the ringbuffer and `test()` relied on `read()`).

Bench testing all checks out.

This PR:
![image](https://user-images.githubusercontent.com/2497951/62649672-7da49a00-b912-11e9-925a-6d23896ab539.png)


PX4:master:
![image](https://user-images.githubusercontent.com/2497951/62649761-a462d080-b912-11e9-940f-b4c1ce2d3257.png)

mcsauder commented:
Hi @julianoes , I've rebased this PR against current master and reverted the changes to the driver usage() method as we discussed in the dev call.  Please let me know if you would like to see any other changes made to this PR!  Thanks!


Console output was tested:
```
teraranger usage
Usage: teraranger <command> [arguments...]
 Commands:

   start         Start driver
     [-a]        Attempt to start driver on all I2C buses (first one found)
     [-b <val>]  Start driver on specific I2C bus
                 default: 1
     [-R <val>]  Sensor rotation - downward facing by default
                 default: 25

   stop          Stop driver

   status        Print driver information
```

julianoes commented on src/drivers/distance_sensor/teraranger/teraranger.cpp:
```diff
@@ -931,91 +558,72 @@ Stop driver
-	PRINT_MODULE_USAGE_COMMAND_DESCR("info","Print driver information");
-
-
+	PRINT_MODULE_USAGE_COMMAND_DESCR("status","Print driver information");
```
Great!

mcsauder commented:
Upstream changes today completely borked the entire master branch history to which I have no explanation.... Merging master became a mess in every branch I have even though I updated them all this morning.  Every last one of my open PR's all became complete disasters.

Here is a new bench test from this PR, it appears to still function: 
![image](https://user-images.githubusercontent.com/2497951/64094060-b2402180-cd17-11e9-8c42-846bd5fd5943.png)

mcsauder commented:
Rebased against current master and bench tested again with pixhawk 4, driver starts automatically when configured to do so and readings all look good:
![image](https://user-images.githubusercontent.com/2497951/64715661-25673780-d47e-11e9-993d-b84101942a19.png)

mcsauder commented:
Rebased on current master to kick the stalled CI tools again.

mcsauder commented:
Thanks @msadowski !  Fixed, rebased, and pushed up!
